GDJS/Runtime/runtimescene.js

```diff
@@ -521,6 +521,14 @@ gdjs.RuntimeScene.prototype.updateObjectsForces = function() {
  * @param obj The object to be added.
  */
 gdjs.RuntimeScene.prototype.addObject = function(obj) {
+    if ( !this._objects.containsKey(obj.name) ) {
+        this._objects.put(obj.name, obj.getOriginalData());
+        this._instances.put(obj.name, []); //Also reserve an array for the instances
+        this._instancesCache.put(obj.name, []); //and for cached instances
+        //And cache the constructor for the performance sake:
+        this._objectsCtor.put(obj.name, obj);
```

Comment:
I'm not sure it's valid to add an instance of a gdjs.RuntimeObject (or a derived class) into a hashtable that should contain constructors. 

That's why I think we should remove this code out of `addObject`. Instead, let the caller code manually register the object with the name, constructor and objectData so it's clear and we're not abusing the _objectsCtor (which is used when creating objects from events).